### ISSUE TITLE
Application loads slow and poor performance

### ISSUE THREAD

**Issue by villesalomaki** [jasontaylordev/CleanArchitecture#165]
**Describe the bug**
Application loads slow and the operation is slow too.

**To Reproduce**
1. git clone https://github.com/jasontaylordev/CleanArchitecture.git
2. Set WebUI as startup project
3. Build the project
4. Press F5
5. After **50 seconds** the main page has loaded. The load time is just as slow if I stop the application and press F5 again.

**Expected behavior**
I expect to see main page loaded well less than 10 seconds.

Also when using the application, and I have not yet registered an account, loading the Todo-page takes 4 seconds.

I first suspected this was somehow related to the authentication. So I created an account and logged in. Then after I have visited each page (I guess them are loaded to browser memory at this point), the performance is better. But if I reload the page, it takes again 4 seconds to load.

**Screenshots**
![image](https://user-images.githubusercontent.com/10339652/84356412-cf6e1e80-abcc-11ea-8ec8-20fc937d02cb.png)

![image](https://user-images.githubusercontent.com/10339652/84356444-ddbc3a80-abcc-11ea-8b47-74ada7c4c686.png)

**Additional context**
It's just not my machine, I asked others and they have the same problem too.

As this is a minimalist reference project, I expect it to be blazing fast, as it's not loading anything extra.

I'm using Windows 10, VS 2019 16.6.2, Resharper, IIS Express, i7-6700HQ 2,60 Ghz, 16 GB RAM, Samsung SSD 970 EVO Plus

Tried latest Chrome and latest Firefox.

**Comment by villesalomaki** [comment]
I get the same bad performance if I create a new project by using the default "ASP.NET Core with Angular" template.
```
dotnet new angular -o TestAngularProject -au Individual
```
Is it really supposed to be this slow? And if yes, why this template was chosen to be used?

**Comment by osama9** [comment]
Try to remove NSwag.

**Comment by litan1106** [comment]
Not on my end with just swagger and the api. (It was blasting fast)
![image](https://user-images.githubusercontent.com/1310539/85304689-e2041580-b460-11ea-95ef-8172473dbec1.png)

**Comment by taylorc** [comment]
@villesalomaki 
tldr; The performance is the dev compilation of Angular not the templates themselves. 

The reason you see "poor" performance on your Dev machine is because of Angular and the default configuration. In the file [Startup.cs](https://github.com/jasontaylordev/CleanArchitecture/blob/master/src/WebUI/Startup.cs) file @ line 123 you have the following:
`              if (env.IsDevelopment())
                {
                    spa.UseAngularCliServer(npmScript: "start");
                }`

The line spa.UseAngularCliServer(npmScript: "start"); has a drawback . Each time you modify your C# code and your ASP.NET Core app needs to restart, the Angular CLI server restarts. The way I bypass this issue is to run the Angular CLI server externally, independently of the ASP.NET Core process. 

1. In a shell, navigate to the ClientApp folder and type  `npm start`. This starts the Angular CLI server independent of Net Core.
2. In the Startup.cs file replace `spa.UseAngularCliServer(npmScript: "start");` with `spa.UseProxyToSpaDevelopmentServer("http://localhost:4200");`

When you start your ASP.NET Core app, it won't launch an Angular CLI server. The instance you started manually is used instead. This enables it to start and restart faster. It's no longer waiting for Angular CLI to rebuild your client app each time.

Within my teams that change helps us add some velocity - also it helps because we use VsCode for front end work such as Angular. 

This dev cycle is one of the primary reason teams choose to use React or Vue over Angular

**Comment by jasontaylordev** [comment]
Thanks @taylorc, this is the preferred approach. I want to keep the default approach for simplicity, but I should add this to the README.
